core/lib/zksync_core/src/api_server/web3/pubsub.rs

```diff
@@ -238,118 +228,143 @@ impl EthSubscribe {
         self.events_sender = Some(sender);
     }
 
-    /// Assigns ID for the subscriber if the connection is open, returns error otherwise.
-    fn assign_id(
-        subscriber: typed::Subscriber<PubSubResult>,
-    ) -> Result<(typed::Sink<PubSubResult>, SubscriptionId), ()> {
-        let id = H128::random();
-        let sub_id = SubscriptionId::String(format!("0x{}", hex::encode(id.0)));
-        let sink = subscriber.assign_id(sub_id.clone())?;
-        Ok((sink, sub_id))
+    async fn reject(sink: PendingSubscriptionSink) {
+        sink.reject(ErrorObject::borrowed(
+            ErrorCode::InvalidParams.code(),
+            "Rejecting subscription - invalid parameters provided.",
+            None,
+        ))
+        .await;
     }
 
-    fn reject(subscriber: typed::Subscriber<PubSubResult>) {
-        subscriber
-            .reject(Error {
-                code: ErrorCode::InvalidParams,
-                message: "Rejecting subscription - invalid parameters provided.".into(),
-                data: None,
-            })
-            .unwrap();
+    async fn subscriber(
+        sink: SubscriptionSink,
+        subscription_type: SubscriptionType,
+        mut b: broadcast::Receiver<Vec<PubSubResult>>,
+        filter: Option<PubSubFilter>,
+    ) -> Result<(), anyhow::Error> {
```

Comment:
Bikeshedding: AFAIU, the error here gets ignored, which IMO looks weird; i.e., errors are used here just for short-circuiting the task. I'd suggest to have the void return value and treat errors in `new_items` and `send_timeout` explicitly. As potential future work, we may meter the number of subscriptions that were closed because of lagging `broadcast::Receiver` and because of a timeout on `send_timeout`; more explicit error handling would help with this.

Reply:
explicit error treating done